File: backend/internal/model/collection.go
package model

import "time"

// CollectionEntry represents a collected card by a user.
type CollectionEntry struct {
	ID         string    `json:"id"`
	UserID     string    `json:"user_id"`
	CardID     string    `json:"card_id"`
	ObtainedAt time.Time `json:"obtained_at"`
	Source     string    `json:"source"` // "game" or "mission"
}

// CollectionStats represents user's collection progress.
type CollectionStats struct {
	UserID      string  `json:"user_id"`
	TotalCards  int     `json:"total_cards"`
	Collected   int     `json:"collected"`
	Percentage  float64 `json:"percentage"`
}

// ObservationMission represents a real-sky observation challenge.
type ObservationMission struct {
	ID        string    `json:"id"`
	CardID    string    `json:"card_id"`
	Title     string    `json:"title"`
	ValidFrom time.Time `json:"valid_from"`
	ValidTo   time.Time `json:"valid_to"`
	Latitude  float64   `json:"latitude"`
	Longitude float64   `json:"longitude"`
}

File: frontend/src/types/collection.ts
export interface CollectionEntry {
  id: string;
  user_id: string;
  card_id: string;
  obtained_at: string;
  source: "game" | "mission";
}

export interface CollectionStats {
  user_id: string;
  total_cards: number;
  collected: number;
  percentage: number;
}

// H4: ObservationMission removed -- unused, missions are in mission.ts
